### PR TITLE
new feature: set arbitrary layer properties

### PR DESCRIPTION
**Pull request by m0nac0**
This introduces a new method: `MaplibreMapController#setLayerProperties` to change any properties of any given layer at runtime.

I had to fix LayerPropertyConverter for android and iOS because there was a problem with the visibility property (which gets somewhat special treatment, and did not work because it received a string like ""visible"" instead of "visible" (notice the unnecessary quotes)). In the process I also fixed the script to generate these files.

TODO:

- [x] update example app to test other properties and layer types

### CODE REVIEW

File: ios/Classes/LayerPropertyConverter.swift
Comment (m0nac0) [inline ios/Classes/LayerPropertyConverter.swift:1]: Note for review: this file was automatically generated by the script.
Apparently it was manually formatted in the past, but I did not do this again.
It is much easier to review when enabling "hide whitespace changes"

File: maplibre_gl_platform_interface/lib/src/source_properties.dart
Comment (m0nac0) [inline maplibre_gl_platform_interface/lib/src/source_properties.dart:1]: also automatically generated

**Comment by m0nac0** [comment]
Closes #250

**Comment by JulianBissekkou** [comment]
@stefanschaller  can you take a look?

File: maplibre_gl_web/lib/src/mapbox_web_gl_platform.dart [hunk 1]
```diff
@@ -881,6 +881,24 @@ class MaplibreMapController extends MapLibreGlPlatform
...
+      try {
+        _map.setLayoutProperty(layerId, entry.key, entry.value);
+      } catch (e) {
+        print('Caught exception (usually safe to ignore): $e');
+      }
+      try {
+        _map.setPaintProperty(layerId, entry.key, entry.value);
+      } catch (e) {
+        print('Caught exception (usually safe to ignore): $e');
+      }
```
Comment (stefanschaller) [inline maplibre_gl_web/lib/src/mapbox_web_gl_platform.dart]:
```suggestion
      try {
        _map.setLayoutProperty(layerId, entry.key, entry.value);
        _map.setPaintProperty(layerId, entry.key, entry.value);
      } catch (e) {
        print('Caught exception (usually safe to ignore): $e');
      }
```

**Review by stefanschaller — APPROVED** [review]

**Comment by stefanschaller** [comment]
Looks good to me. I didn't have any issues with the layer api @JulianBissekkou

File: maplibre_gl_web/lib/src/mapbox_web_gl_platform.dart (hunk 1 above)
Comment (m0nac0) [reply maplibre_gl_web/lib/src/mapbox_web_gl_platform.dart]: I think with your suggestion, if it is a paint property, the setLayoutProperty method would (usually?) throw an error and the setPaintProperty method would not be called.

Comment (m0nac0) [reply maplibre_gl_web/lib/src/mapbox_web_gl_platform.dart]: @stefanschaller What do you think, is it fine to keep the code as is with that explanation?

**Comment by m0nac0** [comment]
@stefanschaller @JulianBissekkou Can we merge this?

**Pull request merged**